util/file.py

```diff
@@ -224,18 +225,25 @@ def mkdir_p(dirpath):
             raise
 
 
-def open_or_gzopen(fname, *opts):
-    return fname.endswith('.gz') and gzip.open(fname, *opts) or open(fname, *opts)
+def open_or_gzopen(fname, *opts, **kwopts):
+    # 'U' mode is deprecated in py3 and may be unsupported in future versions, 
+    # so use newline=None when 'U' is specified
+    for opt in opts:
```

Comment:
you could precede the whole `for` block with `if sys.version_info[0] == 3:` so that the whole block gets skipped under py2 and sys.version_info only gets checked once under py3